hack/unwanted-dependencies.json

```diff
@@ -245,7 +241,6 @@
       "cloud.google.com/go/compute",
       "github.com/GoogleCloudPlatform/k8s-cloud-provider",
       "github.com/gogo/protobuf",
-      "github.com/golang/mock",
```

Comment:
Shouldn't this be kept, to ensure that it doesn't get added back?

Reply:
The “unwantedVendored” section grants exceptions for dependencies which are unwanted but still end up in `vendor`. Since this PR completely removes golang/mock, the exception is no longer necessary, and in fact if the line is kept, dependency validation fails with

```
Good news! Unwanted modules are no longer vendered: ["github.com/golang/mock"]
!!! Remove those from status.unwantedVendored in /home/skitt/go/src/github.com/kubernetes/kubernetes/hack/unwanted-dependencies.json to ensure they don't get reintroduced.
```